### DISCUSSION TITLE
Is it considered safe from a patent perspective?

### DISCUSSION THREAD

**Discussion by StefanOltmann** [awxkee/avif-coder#15] [Q&A] [answered]
HEIC is widely recognized to be heavily encumbered by patents.

Is it safe to incorporate it into a closed-source commercial application, or are licenses required for utilizing this library to decode iPhone HEIC files?

From my observation, it appears to pose significant challenges.

**Comment by awxkee** [comment, marked as answer]
In general I may carefully say yes, because you are not integrating a static code but you're about to use a dynamic library that considered open source `backend`. It should works this way with LGPL 3.0 and many others. One thing is I'm not really sure that `x265` encoder which licensed with GPL 2.0 allow this meaning. Recently `libheif` released supported of open-sourced and free licensed libkvazaar and one time I decided that I'll change x265 to libkvazaar but I didn't

**Reply by StefanOltmann** [reply]
Thank you for your quick response.

I am not concerned about LGPL - I know how to handle that.

My focus is on the HEIC patents. I wonder even if it’s legal for GIMP to have HEIC support.

I just asked the same question on the libheif project to gather more input and views.
Thank you for sharing yours.
